### DISCUSSION TITLE
Sign in failed. Reason: Could not log in with device flow on github.com: self-signed certificate in certificate chain, request id: 5, error code: 1001

### DISCUSSION THREAD

**Discussion by bestsecret28** [community/community#118643] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I am not able to login to co pilot in Intellij. I am using the enterprise licence for co pilot still facing issue on office laptop

**Comment by sumitml** [comment]
Facing the same issue with IntelliJ with enterprise license for co pilot

**Comment by bestsecret28** [comment]
I have solved my issue. The issue is of certificate. Please contact respective admin members to get that certificate and import that certificate in IDE and restart your IDE it will resovle your issue.

This is the process worked for me on my office MAC
1.Create CA Certificate Bundle (With help of admin team.)
2.Move the CA certificate (cacert_combined.pem) into user directory
3.Open ~/.zshrc file and add the command (export NODE_EXTRA_CA_CERTS=“<Path_to_the_cert_bundle>“)
4.Ensure whether the variable (NODE_EXTRA_CA_CERTS) exists in IntelliJ application after adding it in ~/.zshrc file.
5. Invalid cache in the Intellij application and re-open the application Git-hub Co-pilot should work

**Comment by djf-sjtu** [comment]
I have solved the issue, and the process is much more easier!
Just logout from github in your IDE, and sign in github again!

**Comment by hanxiao55** [comment]
I solved the problem after changing to another network. It may be caused by the network firewall. You can try to open the hotspot with your mobile phone.

**Reply by aisspire** [reply]
Thanks a lot bro, I spent almost an hour on this problem until I met you

**Reply by issmatthew** [reply]
Thanks l lot bro,it works.

**Comment by vaechen** [comment]
I encountered a similar problem when logging into copilot in PyCharm.
This is an issue with the network environment. I changed the Settings of PyCharm, setting the HTTP proxy option to automatically detect proxy Settings. Then, I logged in and it took effect.
Use **File > Appearance and behaviour > System setting > HTTP Proxy** in PyCharm. Then click on **Automatic Detection Agent Settings**.

**Comment by romisanic** [comment]
I just solved the issue by closing the copilot's network proxy 
I'm using vpn for my Mac, and in my old version of idea , I configured both system network proxy and github copilot proxy
now i closed the copilot's network proxy, it's working now.path:Copilot -> Edit Settings -> Network.
I tested by opening copilot's network proxy, the error show up agin.
hope helpful.
